modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -157,6 +158,8 @@ const FrontendDataSet = ({
 			}
 		}
 
+		initialActiveView.component = getViewComponent(initialActiveView.name);
+
 		return {
 			activeView: initialActiveView,
```

Comment:
Mutating objects seems weird in React code; I would avoid it.

```suggestion
			activeView: {
			    ...initialActiveView,
			    component: getViewComponent(initialActiveView.name)
			},
```

Reply:
Yeah, that's much better! :+1: